### DISCUSSION TITLE
Can I use only Supabase's storage solution?

### DISCUSSION THREAD

**Discussion by zidkim** [supabase/supabase#6097] [Questions] [answered]
I assume this is the case, but I have to make sure (completely new to supabase/PostgreSQL).

Am I able to use Supabase's storage solution and utilize all its security features without using other solutions (database)?

For example, can I make image download requests from the client without Supabase's authentication in place?
I'm using Firebase auth at the moment.
My goal is to utilize both Firebase and Supabase for different parts of my app.

Thanks in advance!

**Comment by GaryAustin1** [comment, marked as answer]
I'm not sure how easy this is going to be with out Supabase auth if you want security.  I've not seen anything on this.

Storage API requires a jwt with a least a sub field.
This getOwner function is used throughout to get the userId (or null in the case of no user, or serviceKey)
```
export async function getOwner(token: string, secret: string): Promise<string | undefined> {
  const decodedJWT = await verifyJWT(token, secret)
  return (decodedJWT as jwtInterface)?.sub
}
```
Without a user ID set in object owner field, and then a jwt passed in on downloads (etc.), RLS won't be able to do much.

So if you if by security you meant storage RLS policy, I think you have more research to do on how to deal with the JWT situation without Supabase Auth, or to duplicate users in both platforms...

**Reply by zidkim** [reply]
Understood, thank you.

Have you seen solutions where we authenticate with Supabase, and then use that auth state across both Supabase and Firebase?
I imagine this to be possible -- here's an example of what I mean.

1. Authenticate with one of Supabase auth solutions.
2. Pass JWT (custom token) to [Firebase Custom Authentication System](https://firebase.google.com/docs/auth/web/custom-auth#authenticate-with-firebase).
3. Create the 'user' object with all its properties and save it to Firebase and Supabase in server (Firebase cloud function in my case).
4. Say I make a request to firebase, and on that document there's a storage/download url to Supabase's storage. This would result in success, as the userId/uuid is shared across platforms. Correct?

On a side note, Supabase storage solution is layered with its own CDN, correct?

Last side note -- I see that on Supabase pricing page, there is "Transfer limits" in both Database and Storage sections. The transfer limit (download/bandwidth) is in relation to the Database API and the Storage API respectively, yes? If you think I should open another ticket for this, I'll go ahead and do that.

**Reply by GaryAustin1** [reply]
There are other users who are dealing with Firebase and Supabase and an auth connection that I have seen, but I would have to google, or search here in discussions and out of time right now.  One for sure was creating Supabase users to match Firebase existing users.

Storage does go thru a CDN.

There are transfer download limits for both storage and database, and charges for higher levels.  My personal assumption is this includes CDN downloads as CDN's normally have a charge.

This is not really a direct support forum, but users and some Supabase employees (I'm not one) helping out.   You mentioned "ticket" which makes me think of support.

**Reply by zidkim** [reply]
Ah, thank you again.
Big help.
